lib/analyzer/applications/java.ts

```diff
@@ -0,0 +1,56 @@
+import { JarFingerprint } from "../types";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+function mapContentToJarFingerprints(input: {
+  [fileName: string]: string;
+}): {
+  [path: string]: JarFingerprint[];
+} {
+  const jarFingerprints: JarFingerprint[] = Object.entries(input).map(
+    ([key, value]) => {
+      const path = key.substring(0, key.lastIndexOf("/"));
```

Comment:
I think it's better to handle this with the `path` module as it can deal with `/` vs `\\` in paths

Reply:
ah great point, thanks!!